crates/phala-types/src/lib.rs

```diff
@@ -681,6 +683,54 @@ pub struct WorkerRegistrationInfo<AccountId> {
     pub operator: Option<AccountId>,
 }
 
+#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, TypeInfo, Hash)]
+pub enum EndpointType {
+    I2P = 0,
+    Http,
+}
+
+impl FromStr for EndpointType {
+    type Err = ParseError;
+    fn from_str(endpoint_type: &str) -> Result<Self, Self::Err> {
+        match endpoint_type {
+            "i2p" => Ok(EndpointType::I2P),
+            "http" => Ok(EndpointType::Http),
+            _ => Ok(EndpointType::I2P), // default to I2P
+        }
+    }
+}
+
+#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, TypeInfo)]
+pub enum VersionedWorkerEndpoints {
+    V1(Vec<worker_endpoint_v1::EndpointInfo>),
+}
+
+#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, TypeInfo)]
+pub struct SignedWorkerEndpoint {
```

Comment:
You cannot call it "signed" unless it has the signature attached. Good to call it "WorkerEndpointPayload".

Reply:
Done